storage/fanout.go

```diff
@@ -240,23 +240,43 @@ func (q *mergeQuerier) Select(params *SelectParams, matchers ...*labels.Matcher)
 func (q *mergeQuerier) SelectSorted(params *SelectParams, matchers ...*labels.Matcher) (SeriesSet, Warnings, error) {
 	seriesSets := make([]SeriesSet, 0, len(q.queriers))
 	var warnings Warnings
+
+	var priErr error = nil
+	type queryResult struct {
+		qr          Querier
+		set         SeriesSet
+		wrn         Warnings
+		selectError error
+	}
+	queryResultChan := make(chan *queryResult)
 	for _, querier := range q.queriers {
-		set, wrn, err := querier.SelectSorted(params, matchers...)
-		q.setQuerierMap[set] = querier
-		if wrn != nil {
-			warnings = append(warnings, wrn...)
+		go func(qr Querier) {
+			set, wrn, err := qr.SelectSorted(params, matchers...)
+			queryResultChan <- &queryResult{qr: qr, set: set, wrn: wrn, selectError: err}
+		}(querier)
+	}
+	for i := 0; i < len(q.queriers); i++ {
+		qryResult := <-queryResultChan
+		if qryResult.wrn != nil {
```

Comment:
You can simplify by removing this and the next `if`. The performance cost will be negligible.

Reply:
done.Thank you for your review.
![image](https://user-images.githubusercontent.com/9583245/74256787-e3913480-4d2e-11ea-899e-d35ade8101b4.png)